### DISCUSSION TITLE
new f-method .as()

### DISCUSSION THREAD

**Discussion by st-pasha** [h2oai/datatable#2684] [Ideas]
I think our current API lacks somewhat when it comes to renaming columns in the j section. The "normal" API is to use a dict: `{"newA": f.A, "newB": f.B, ...}`. However, a common use-case is when you want to select some columns as-is, and also some new columns with names. In this case repeating names of all columns may be too difficult or even impossible.

A possible workaround is to use the `.extend()` method. However, this has its own drawbacks:
- it is quite non-obvious;
- the resulting expression is still very complicated;
- if you start with an expression such as `DT[:, [f.A, f.B, f.C]]` and then decide that you need to rename, say, column `B`, then you end up having to completely rewrite the formula into `DT[:, f.A.extend({"newB": f.B}).extend(f.C)]`.

The solution that I'm proposing is to have a new method `.as()` that could rename an individual f-expression. For example:
```
DT[:, [f.A, f.B.as("newB"), f.C]]
```
(the name is inspired by the equivalent SQL keyword).

Similarly, you can apply this to a columnset and rename multiple columns at once:
```
DT[:, f[:2].as("Price", "Quantity")]
```

**Comment by pradkrish** [comment]
That is a good idea. Also, modifiers - `by(),` `sort()` and `join()` - within the square-brackets should be able to operate on the alias.

**Comment by samukweku** [comment]
this has been resolved with the `.alias` method
